### ISSUE TITLE
Testing methods

### ISSUE THREAD

**Issue by sbrooke** [mmdoogie/roku-imgserver-screensaver-server#2]
Should I be able to open the imgserver from a browser and get a result? Trying to confirm it's not a problem with my Roku configuration.

Also, will the program handle an empty API variable? I don't have one. At least, I don't think I do.

**Comment by mmdoogie** [comment]
imgserver.php?cmd=descs should show the configured imgserver.php?cmd=descs should show the configured descriptions

imgserver.php?cmd=img&num=0 should show the first image

Where is the API variable you mentioned? I don’t think this uses any APIs.

**Comment by sbrooke** [comment]
API is in the do-fetch file. "APITOKEN="

I'm getting an empty image when I try to show the first image. I'll try some other things later today.

**Comment by mmdoogie** [comment]
Ohhhh yes, I need to add that to the instructions as well. You’ll need to add an API token or service account in your Grafana settings so it has a way to log in. You also need to have the Grafana Image Renderer plug-in installed if it’s not already.

**Comment by sbrooke** [comment]
I have Grafana and GIR running in separate dockers on another system. I can pull an anon kiosk image from the GIR without issue. As long as the dashboard is set up for anonymous access you shouldn't need an API token.

**Comment by mmdoogie** [comment]
You could just use the non grafana fetcher version and use the full http url to what you know works to fetch the image.

**Comment by sbrooke** [comment]
Yeah, I tried that as well. It takes longer, like the GIR is rendering the image, but it's still an empty file. 275 bytes or something. I can run the same URL in a regular browser and I get a result from the GIR, so I know the URL formatting is correct.

Could this be permissions related? Is the script failing to write properly? I see the img.png file in the server so it's writing it there, but it's also only 275 bytes.

**Comment by mmdoogie** [comment]
Imagemagick is installed on the one with PHP and the convert command is available?

**Comment by sbrooke** [comment]
I get this with "convert -version"

`Version: ImageMagick 6.9.11-60 Q16 x86_64 2021-01-25 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC Modules OpenMP(4.5) 
Delegates (built-in): bzlib djvu fftw fontconfig freetype heic jbig jng jp2 jpeg lcms lqr ltdl lzma openexr pangocairo png tiff webp wmf x xml zlib`

**Comment by mmdoogie** [comment]
Comment out the rm and see if the image it downloads with curl to tmpImg is able to be written?

**Comment by sbrooke** [comment]
Ok, I've established that having &kiosk in the url breaks it. It kind of looks like any variables in the url breaks it. So, the base image url works but then it's formatted to the wrong size.

Sorry for all the trouble!

**Comment by mmdoogie** [comment]
I use variables with some of mine, first one needs a ? though for proper url syntax

**Comment by sbrooke** [comment]
Good news, bad news. If I add `?kiosk` it works. If I add `?kiosk&anythingElse` it fails. I've tried multiple cmds after the kiosk and they all fail. I also tried it on the Roku itself again, and I get nothing on there, no matter how the url is formatted.

**Comment by mmdoogie** [comment]
Oh yep that sounds like a bug

https://github.com/mmdoogie/roku-imgserver-screensaver-server/blob/d0b5d542e599e4349dd96f17975926b4537dc2cb/imgserver.php#L61

On this line, put \” before and after $url

I believe that will fix it but I’m not at my computer to try it at the moment.

**Comment by sbrooke** [comment]
That fixed it for the test download! Not working on Roku. I'm checking if that's something I did wrong.

**Comment by mmdoogie** [comment]
How long does it take to render? If it’s more than 5-10 seconds it’ll time out and I don’t think there’s a way to extend that.

**Comment by sbrooke** [comment]
Only takes ~3 seconds to render. My problem was url formatting in the Roku app. All good now! The image formatting is a little weird, but that's probably just some tweaking for my TV. Thanks for being responsive on all of this! I've been running an X Window session on a RPi to display the Grafana dashboard. It's not without it's own issues. Having the built in Roku on the TV will help a lot!

**Comment by mmdoogie** [comment]
Which Roku app are you using?

**Comment by sbrooke** [comment]
The one you created. Using it through the developer upload.

**Comment by mmdoogie** [comment]
Oh ok just throw me I wasn’t sure how the formatting would be different

**Comment by sbrooke** [comment]
Roku Express on an old Samsung 720p TV. It was doing overscan and going out past the edges of the screen. I also need to test the different display modes. 1080p dashboard to a 720p screen? Does that get too pixelated? Does it even display right? Just issues specific to my situation.

**Comment by mmdoogie** [comment]
You can look at the renderconf parameters in the grafana fetch one that shows how to ask for different sizes. I render mine at 2560x1440 and scale down to 720p and it’s still quite readable.

**Comment by sbrooke** [comment]
I'm using the plain, non-Grafana do-fetch and it appears to not be displaying it if the screen resolution doesn't match the cmd switches in the url string. ie, if I set the url to 1920x1080 and the Roku is set to 720p display it will not show an image. That's a minor issue, and one that's easily worked around.

**Comment by mmdoogie** [comment]
The convert command down/up scales whatever is returned from the image to the Roku resolution so if it’s not giving an image it’s not getting one. I use it with resolutions that are all over the place on both 720p and 1080p TVs with no changes.
